tasks.py

```diff
@@ -1,18 +1,54 @@
 import asyncio
 
+import httpx
+from lnbits.core.crud import get_wallet
 from lnbits.core.models import Payment
 from lnbits.core.services import (
     create_invoice,
     get_pr_from_lnurl,
     pay_invoice,
     websocket_updater,
 )
+from lnbits.helpers import create_access_token
+from lnbits.settings import settings
 from lnbits.tasks import register_invoice_listener
 from loguru import logger
 
 from .crud import get_tpos
 
 
+async def _deduct_inventory_stock(wallet_id: str, inventory_payload: dict) -> None:
+    wallet = await get_wallet(wallet_id)
+    if not wallet:
+        return
+    inventory_id = inventory_payload.get("inventory_id")
+    items = inventory_payload.get("items") or []
+    if not inventory_id or not items:
+        return
+    ids: list[str] = []
+    quantities: list[int] = []
+    for item in items:
+        item_id = item.get("id")
+        qty = item.get("quantity") or 0
+        if not item_id or qty <= 0:
+            continue
+        ids.append(item_id)
+        quantities.append(int(qty))
+    if not ids:
+        return
+    # Needed to accomodate admin users, as using user ID is not possible
+    access = create_access_token(
+        {"sub": "", "usr": wallet.user}, token_expire_minutes=1
+    )
+    async with httpx.AsyncClient() as client:
+        await client.patch(
+            url=f"http://{settings.host}:{settings.port}/inventory/api/v1/items/{inventory_id}/quantities",
```

Comment:
```suggestion
            url=f"inventory/api/v1/items/{inventory_id}/quantities",
```

no need for host,port

Reply:
it does! httpx needs the full path!

the solution is not ideal, but i can't see any workaround!